### PR TITLE
chore: improve test coverage.

### PR DESCRIPTION
**Pull request by jparsai**
#### Description:
This PR is to add unit test cases for gitopsdeploymentrepositorycredential_types.go file of backend-shared.

#### JIRA Ticket:
https://issues.redhat.com/browse/GITOPSRVCE-690

### CODE REVIEW

File: backend-shared/apis/managed-gitops/v1alpha1/gitopsdeploymentrepositorycredential_types_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+			Expect(status.Conditions[0].LastTransitionTime).NotTo(Equal(temp.LastTransitionTime))
+		})
+
+		It("Should add new conditions and sort them entry in Status.Conditions field.", func() {
+
+			status := GitOpsDeploymentRepositoryCredentialStatus{}
+			Expect(status.Conditions).To(BeEmpty())
+
+			conditions := []metav1.Condition{
```
Comment (keithchong) [inline backend-shared/apis/managed-gitops/v1alpha1/gitopsdeploymentrepositorycredential_types_test.go:121]: Could some of these steps be put in the BeforeEach clause? This conditions value is different though compared to the previous three.

Comment (jparsai) [reply backend-shared/apis/managed-gitops/v1alpha1/gitopsdeploymentrepositorycredential_types_test.go]: I have changed it, PTAL.

**Review by keithchong — APPROVED** [review]
LGTM

**Pull request merged**